src/lib/repository/DeviceRepository.js

```diff
@@ -138,9 +143,16 @@ class DeviceRepository {
 
   callFunction= async (
     deviceID: string,
+    userID: string,
     functionName: string,
     functionArguments: Object,
   ): Promise<*> => {
+    const doesCoreBelongsToUser =
```

Comment:
i made this, but i'm not sure if I like it.

 const core = this._deviceServer.getCore(deviceID); - doesn't support userId checks, and perhaps it shouldn't....so I checked it by getting deviceAttributes with deviceID and userID